Assets/Scripts/Crab.cs

```diff
@@ -0,0 +1,154 @@
+        _facingDirection = RIGHT;
+        _rb2d = GetComponent<Rigidbody2D>();
+        _anim = GetComponent<Animator>();
+
+        // kad nepermestu i game ekrana, o liktu scene
+        if (keepSceneViewActive && Application.isEditor)
+        {
+            UnityEditor.SceneView.FocusWindowIfItsOpen(typeof(UnityEditor.SceneView));
+        }
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    private void FixedUpdate()
+    {
+        float vX = _moveSpeed;
+
+        if(_facingDirection == LEFT)
+        {
+            vX = -_moveSpeed;
+        }
+
+
+        _rb2d.velocity = new Vector2(vX, 0);
+
+        if (IsPlayerClose())
+        {
+            MoveTowardsPlayer();
+        }
+
+        if(IsPlayerInAttackRange())
+        {
+            Attack();
+        }
+
+        if (IsHittingWall() || IsAboutToFall())
+        {
+            ChangeFacingDirection();
+        }
+    }
+
+    private void ChangeFacingDirection()
+    {
+        Vector3 currentScale = transform.localScale;
+        currentScale.x = -transform.localScale.x;
+        transform.localScale = currentScale;
+
+        _facingDirection = _facingDirection == LEFT ? RIGHT : LEFT;
+    }
+
+    private void ChangeFacingDirection(string direction)
+    {
+        if(_facingDirection != direction)
+        {
+            ChangeFacingDirection();
+        }
+    }
+
+    private bool IsHittingWall()
+    {
+        float castDist = _facingDirection == LEFT ? -_baseHorizontalCastDist : _baseHorizontalCastDist;
+
+        Vector3 targetPos = _castPos.position;
+        targetPos.x += castDist;
+
+        Debug.DrawLine(_castPos.position, targetPos, Color.green);
+
+        bool val = Physics2D.Linecast(_castPos.position, targetPos, 1 << LayerMask.NameToLayer("Ground"));
+
+        return val;
+    }
+
+    private bool IsAboutToFall()
+    {
+        Vector3 targetPos = _castPos.position;
+        targetPos.y -= _baseVerticalCastDist;
+
+        Debug.DrawLine(_castPos.position, targetPos, Color.green);
+
+        bool val = !Physics2D.Linecast(_castPos.position, targetPos, 1 << LayerMask.NameToLayer("Ground"));
+
+        return val;
+    }
+
+    private bool IsPlayerClose()
+    {
+        // crab and player are on different levels vertically so shouldn't follow even if it's close
+        var verticalFollowDist = 0.6f;
+        if (Math.Abs(_playersTransform.position.y - _rb2d.transform.position.y) > verticalFollowDist)
+        {
+            return false;
+        }
+
+        return Vector2.Distance(_playersTransform.position, _rb2d.transform.position) < playerFollowDist;
+    }
+
+    private void MoveTowardsPlayer()
+    {
+        ChangeFacingDirection(_rb2d.transform.position.x < _playersTransform.position.x ? RIGHT : LEFT);
+    }
+
+    private bool IsPlayerInAttackRange()
+    {
+        var attackRange = 0.6f;
+        return Vector2.Distance(_playersTransform.position, _rb2d.transform.position) < attackRange;
```

Comment:
Since there won't be many mobs in a single level this approach is okay, but it can be changed to: 
`return (_playersTransform.position - _rb2d.transform.position).sqrMagnitude < attackRange * attackRange`
Because Vector2.Distance uses sqr operation which is expensive and not needed in this case.